packages/nimble-components/src/rich-text/editor/testing/rich-text-editor.pageobject.ts

```diff
@@ -98,6 +98,24 @@ export class RichTextEditorPageObject {
         await waitForUpdatesAsync();
     }
 
+    // In testing environment, when clicking on the footer button, it may not persist in the same state if any editor transaction occurs in between.
+    // This behavior is likely due to dynamic modifications of formatting button states based on cursor positions during editor transactions.
+    // By setting the "force" parameter to true in our method, will ensure the attainment of our intended state by interacting with it twice.
+    public async toggleFooterButton(
```

Comment:
I think @rajsite's proposal was to make `force` behave like browser APIs [like this one](https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute#force).

- if not specified at all, "toggles" the value by clicking it once
- if true, sets the value to true
- if false, sets the value to false

For the true and false cases we'd click the button once if it needs to change its value. If it's already at the current value we could either do nothing or click it twice if that's required by the quirk you mention in your comment.

Once we do that, this method could replace the `clickFooterButton()` method since it would have identical behavior as calling this with `force` unset. You could delete that method in this PR or [mark it deprecated](https://jsdoc.app/tags-deprecated) and remove it in a future PR.

Reply:
Updated it. If `force` sets true/false will check the button state and update it based on its active state. If unset it will just click one time (same behavior).

Reply:
Since all usages of clickFooterButton seem to be in this file please go ahead and switch to toggleFooterButton. It's literally just a find and replace.

I don't see clickFooterButton used here or [in azdo](https://dev.azure.com/ni/_search?text=clickFooterButton&type=code&lp=custom-Collection&filters=&pageSize=25) so I say we should remove it instead of deprecating it.

Reply:
Removed `clickFooterButton` method